hip_data_tools/common.py

```diff
@@ -62,3 +63,129 @@ def get_from_env_or_default_with_warning(env_var, default_val):
         value = default_val
 
     return value
+
+
+class KeyValueSource(ABC):
+    """
+    Abstract class for sourcing secrets, it is a key value source for retrieving values for keys
+    """
+
+    @abstractmethod
+    def get(self, key: str) -> str:
+        """
+        Abstract method to get the value for a given key
+        Args:
+            key (str): the key for which the value needs to be returned
+        Returns: str
+        """
+        pass
+
+    @abstractmethod
+    def exists(self, key):
+        """
+        Abstract methosd to verify if a key exists in the given data store
+        Args:
+            key (str): the key to be verified for existance
+        Returns: bool
+        """
+        pass
+
+
+class EnvironmentKeyValueSource(KeyValueSource):
+    """
+    class for sourcing secrets from env variables
+    """
+
+    def exists(self, key):
+        """
+        verify if a key exists
+        Args:
+            key (str): the key to be verified for existance
+        Returns: bool
+        """
+        if os.getenv(key) is None:
+            return False
+        return True
+
+    def get(self, key):
+        """
+        get the value for a given key
+        Args:
+            key (str): the key for which the value needs to be returned
+        Returns: str
+        """
+        return os.getenv(key)
+
+
+class DictKeyValueSource(KeyValueSource):
+    """
+    class for sourcing secrets from a provided Dict object, usually used for testing
+    """
+
+    def __init__(self, data):
+        self.data = data
+
+    def exists(self, key):
+        """
+        verify if a key exists
+        Args:
+            key (str): the key to be verified for existance
+        Returns: bool
+        """
+        if key in self.data:
+            return True
+        return False
+
+    def get(self, key):
+        """
+        get the value for a given key
+        Args:
+            key (str): the key for which the value needs to be returned
+        Returns: str
+        """
+        return self.data[key]
+
+
+ENVIRONMENT: EnvironmentKeyValueSource = EnvironmentKeyValueSource()
```

Comment:
does this need to be created at the module level? 
if it does it should be commented and described below or above?

Reply:
dones